File: Backend/SeatifyBackend/Entities/Models/Appearance.cs
using System.ComponentModel.DataAnnotations;
using Entities.Helpers;

namespace Entities.Models
{
    public class Appearance : IIdEntity
    {
        [Key]
        public string Id { get; set; } = Guid.NewGuid().ToString();

        public string OrganizerId { get; set; } = string.Empty;

        public Organizer Organizer { get; set; } = null!;

        public string PrimaryColor { get; set; } = string.Empty;
        
        public string SecondaryColor { get; set; } = string.Empty;
        
        public string LogoImageUrl { get; set; } = string.Empty;
        
        public string BannerImageUrl { get; set; } = string.Empty;
        
        public string ThemePreset { get; set; } = "Default (Blue)";

        public bool IsDefault { get; set; } = false;

        public DateTime CreatedAtUtc { get; set; }

        public DateTime UpdatedAtUtc { get; set; }
    }
}

File: Frontend/SeatifyFrontend/src/app/models/appearance.ts
export interface Appearance {
  id: string;
  organizerId: string;
  name: string;
  primaryColor: string;
  accentColor: string;
  backgroundColor: string;
  surfaceColor: string;
  textColor: string;
  secondaryColor: string;
  logoImageUrl: string;
  bannerImageUrl: string;
  themePreset: string;
  fontFamily: string;
  isDefault: boolean;
  createdAtUtc: string;
  updatedAtUtc: string;
}

export interface AppearanceCreateRequest {
  name: string;
  primaryColor: string;
  accentColor: string;
  backgroundColor: string;
  surfaceColor: string;
  textColor: string;
  secondaryColor: string;
  logoImageUrl: string;
  bannerImageUrl: string;
  themePreset: string;
  fontFamily: string;
  isDefault: boolean;
}
